velox/functions/prestosql/MapTopN.h

```diff
@@ -34,7 +34,13 @@ struct MapTopNFunction {
           CompareFlags::NullHandlingMode::kNullAsIndeterminate};
 
       if (l->second.has_value() && r->second.has_value()) {
-        return l->second.value().compare(r->second.value(), flags) > 0;
+        auto comp = l->second.value().compare(r->second.value(), flags);
+
+        if (FOLLY_UNLIKELY(comp == 0)) {
+          return l->first.compare(r->first, flags) > 0;
```

Comment:
What happens if keys are not orderable?

Reply:
Right now Presto throws an exception if you have keys that are not orderable : 
```
SELECT
    MAP_TOP_N(
        MAP(
            ARRAY[
                MAP(ARRAY[1], ARRAY[2]),
                MAP(ARRAY[3], ARRAY[2]),
                MAP(ARRAY[14], ARRAY[2])],
                ARRAY[1, 2, 3]),3 )
   ```
   
   I will validate and add test cases for that.

Reply:
Changed signature to ensure keys are orderable.